search_knn.go

```diff
@@ -349,10 +342,35 @@ func mergeKNNDocumentMatches(req *SearchRequest, knnHits []*search.DocumentMatch
 // hits, then the top K hits need to be distributed to I1 and I2,
 // so that the preSearchData for I1 contains the top K hits from I1 and
 // the preSearchData for I2 contains the top K hits from I2.
-func distributeKNNHit(docMatch *search.DocumentMatch, indexNumToDocMatchList map[int][]*search.DocumentMatch) {
-	top := docMatch.IndexId[len(docMatch.IndexId)-1]
-	docMatch.IndexId = docMatch.IndexId[:len(docMatch.IndexId)-1]
-	indexNumToDocMatchList[top] = append(indexNumToDocMatchList[top], docMatch)
+func distributeKNNHits(knnHits []*search.DocumentMatch) map[string][]*search.DocumentMatch {
+	segregatedKnnHits := make(map[string][]*search.DocumentMatch)
+	for _, hit := range knnHits {
+		stackTopIdx := len(hit.IndexNames) - 1
+		top := hit.IndexNames[stackTopIdx]
+		hit.IndexNames = hit.IndexNames[:stackTopIdx]
+		segregatedKnnHits[top] = append(segregatedKnnHits[top], hit)
+	}
+	return segregatedKnnHits
+}
+
+func checkIndexNamesConsistency(knnHits []*search.DocumentMatch, indexes []Index) error {
+	// create a set of all the index names of this alias
+	indexNames := make(map[string]bool)
+	for _, index := range indexes {
+		indexNames[index.Name()] = true
+	}
+
+	// a topology change / inconsistency is detected when, for a given KNN hit,
```

Comment:
I don't think we need this commentary. 
Code is simple enough and self explanatory .